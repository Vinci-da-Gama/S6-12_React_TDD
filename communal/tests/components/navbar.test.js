import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import NavbarCompo from '../../src/components/navbar';

test('should render navbar correctly.', () => {
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<NavbarCompo />);
    // expect(renderer.getRenderOutput()).toMatchInlineSnapshot();
    const wrapper = shallow(<NavbarCompo />);
    expect(wrapper).toMatchSnapshot();
});
