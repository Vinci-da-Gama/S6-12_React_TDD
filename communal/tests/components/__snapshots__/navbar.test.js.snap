// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render navbar correctly. 1`] = `
<Navbar
  color="light"
  expand="md"
  light={true}
  tag="nav"
>
  <NavbarBrand
    activeClassName="active"
    exact={true}
    tag={[Function]}
    to="/"
  >
    Show_Case
  </NavbarBrand>
  <NavbarToggler
    onClick={[Function]}
    tag="button"
    type="button"
  />
  <Collapse
    appear={false}
    enter={true}
    exit={true}
    in={false}
    isOpen={false}
    mountOnEnter={false}
    navbar={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    tag="div"
    timeout={350}
    unmountOnExit={false}
  >
    <Nav
      className="mx-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          exact={true}
          tag={[Function]}
          to="/"
        >
          Expense_Dashboard
        </NavLink>
      </NavItem>
      <UncontrolledDropdown
        inNavbar={true}
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          caret={true}
          color="secondary"
          nav={true}
        >
          Expense_CRUD
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <DropdownItem
            tag="button"
            toggle={true}
          >
            <NavLink
              activeClassName="active"
              tag={[Function]}
              to="/create"
            >
              Create
            </NavLink>
          </DropdownItem>
          <DropdownItem
            divider={true}
            tag="button"
            toggle={true}
          />
          <DropdownItem
            tag="button"
            toggle={true}
          >
            <NavLink
              activeClassName="active"
              tag={[Function]}
              to="/help"
            >
              Help
            </NavLink>
          </DropdownItem>
        </DropdownMenu>
      </UncontrolledDropdown>
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          tag={[Function]}
          to="/inebriant"
        >
          Inebriant
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          tag={[Function]}
          to="/health"
        >
          Health
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          tag={[Function]}
          to="/cosmetic_hlDegate"
        >
          Cosmetic_N_HighLevelCompoDelegate
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          tag={[Function]}
          to="/nestedInDecisionNCounter"
        >
          InDecision_And_Counter
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          tag={[Function]}
          to="/nestedBooksByParams"
        >
          Books_With_Params_Id
        </NavLink>
      </NavItem>
    </Nav>
  </Collapse>
</Navbar>
`;
